Various adjustments to create-extension

ia3andy opened the pull request:
and:
- fix pom.xml missing quarkus version
- fix doc -> docs in quarkiverse

ia3andy committed:
Use integration-tests as convention & fix pom.xml

gastaldi reviewed (approved):
LGTM

gsmet commented on independent-projects/tools/devtools-common/src/main/resources/codestarts/quarkus-extension/code/integration-tests/java/integration-tests/pom.tpl.qute.xml:
```diff
@@ -49,6 +49,9 @@
                 <artifactId>quarkus-maven-plugin</artifactId>
+                {#if quarkus.version}
+                <version>$\{quarkus.version}</version>
+                {/if}
```
I would rather add it in the `pluginManagement` of the parent.

ia3andy replied:
@gsmet I am not sure as the `quarkus-maven-plugin` is only used by the `integration-tests`. Is there a case where we use it in runtime or deployment?

gsmet replied:
Yes but you really don't know what the user will do and better keep it consistent. Having it in `pluginManagement` won't execute it, it will just make it the proper version is used everywhere it is declared.

gsmet reviewed (changes requested):
Added one small gripe.

ia3andy committed:
Use docs instead of doc for quarkiverse extensions

ia3andy committed:
Use version param for quarkiverse .github/project.yml version

gsmet committed:
Remove properties that shouldn't be in the extension IT template

It's better to not have them here as the extension might need to add
build items so that they are properly propagated to any app containing
the extension.

ia3andy committed:
Fix snapshot testing library when snapshot doesn't exist
By default, it should fail to avoid succeeding in CI when the snapshot doesn't exist

ia3andy commented:
is it me or the CI is getting unstable again?
